subprojects/frameworklauncher/yarn/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -348,6 +366,14 @@ private void stopForInternalUnKnownError(String customizedDiagnostics) {
     stop(new StopStatus(ExitStatusKey.AM_INTERNAL_UNKNOWN_ERROR.toInt(), false, diagnostics));
   }
 
+  private void stopForApplicationStopRequest(String customizedDiagnostics) {
+    String diagnostics = ExitDiagnostics.generateDiagnostics(
+        ExitStatusKey.LAUNCHER_STOP_FRAMEWORK_REQUESTED, customizedDiagnostics);
+
+    // Do not unregister to treat it conservatively.
+    stop(new StopStatus(ExitStatusKey.LAUNCHER_STOP_FRAMEWORK_REQUESTED.toInt(), true, diagnostics));
+  }
```

Comment:
LAUNCHER_STOP_FRAMEWORK_REQUESTED is defined to be NON_TRANSIENT, so we should unregister

Reply:
We do.

Reply:
OK but you forget to change the comment.